### DISCUSSION TITLE
Release version 2.0 2024-11-17

### DISCUSSION THREAD

**Discussion by lcn2** [ioccc-src/mkiocccentry#1030] [Announcements]
## Release 2.0 2024-11-17

This is a formal release for public use.

Changed `MKIOCCCENTRY_REPO_VERSION` from "1.6.13 2024-11-16"
to "2.0 2024-11-17".

This repo is now under a **CODE FREEZE**.

Only super-critical bugs that have a significant impact the running
of IOCCC28 or the **Great Fork Merge** process and the temp-test-ioccc
repo will be considered until sometime after the winners of IOCCC28 have
been released.

## Release 1.6.14 2024-11-17

Update from dbg and jparse repos.

Prep for code freeze and formal release for IOCCC28.

## Release 1.6.13 2024-11-16

Added minor changes to the `jparse/jstrdecode(1)` tool.

## Release 1.6.12 2024-11-16

Add missing `-q` option to `txzchk`.

## Release 1.6.11 2024-11-15

Fixes and improvements in `jparse` synced from the [jparse
repo](https://github.com/xexyl/jparse/). This includes an update to the jparse
library and some bug fixes and additions to the jstrdecode and jstrencode tools.

Typo fix in mkiocccentry.c.

Also roll back a commit in chkentry.c - a work in progress for **after** IOCCC28
that was, in a rush to leave the house but still get important fixes in, was
accidentally committed yesterday.

## Release 1.6.10 2024-11-14

Synced `jparse/` from the [jparse repo](https://github.com/xexyl/jparse/). The
term `encode` and `decode` were swapped back to the original meaning. This
problem occurred due to the fact that when focusing for some time on the
encoding/decoding bug of `\uxxxx` code points in the jparse repo the fact that
it is a **JSON** encoder was lost in focus. This change also means that terms
were swapped here.

In the process of an issue in 'the other repo', some bugs were uncovered in
`jstrencode(1)` but this should not, I believe, affect the timeline of the next
IOCCC, fortunately. The website tools have to be updated which I will work on
next a bit later on today.

## Release 1.6.9 2024-11-09

Synced `jparse/` from the [jparse repo](https://github.com/xexyl/jparse/). This
came from discussion on JSON debug output as well as discussion about wording. A
fix in one of the man pages was also made (no longer true statement) and the man
pages and utils README.md were all expanded with more details and examples.

## Release 1.6.8 2024-11-08

Synced `jparse/` from the [jparse repo](https://github.com/xexyl/jparse/). This
includes some important bug fixes in a utility function that resulted, in debug
output, invalid JSON, plus an incorrect calculation in one case.

## Release 1.6.7 2024-11-07

Synced `jparse/` from the [jparse repo](https://github.com/xexyl/jparse/) to
remove unused code and extra sanity checks added to a function and to match a
filename change.

Improve (to help parse output a bit better) the scripts `bug_report.sh` and
`hostchk.sh`.

## Release 1.6.6 2024-11-05

Synced `jparse/` from the [jparse repo](https://github.com/xexyl/jparse/) with
some important documentation changes along with some typo fixes.

## Release 1.6.5 2024-11-03

Sync the `jparse/` directory from [jparse
repo](https://github.com/xexyl/jparse/). This adds some extra sanity checks in
`jstrencode(1)`/`jstrdecode(1)` and it also makes some fixes to the change in
decode/encode terminology, namely to do with comments. An unused macro (after a
fix) was removed. Comments were updated and improved in `jparse.l` and
`jparse.y`. Regenerated parser backup files.

## Release 1.6.4 2024-11-01

Sync the `jparse/` directory from [jparse
repo](https://github.com/xexyl/jparse/). This includes some important bug fixes,
namely that some codepoints that were rejected are no longer rejected as later
Unicode standards allow them. This means that the
[JSONTestSuite](https://github.com/nst/JSONTestSuite) files now completely pass
jparse.

The `CODE_OF_CONDUCT.md` was removed from `jparse/` as it referred to the IOCCC
judges which is not correct. Until a better way of doing this is derived, this
file no longer exists. To be clear the mkiocccentry `CODE_OF_CONDUCT.md` still
exists!

## Release 1.6.3 2024-10-31

Sync the `jparse/` directory from [jparse
repo](https://github.com/xexyl/jparse/). This includes some important bug fixes,
including a segfault when JSON debug level was specified and a string was
encountered (the problem was that when `has_nul` was removed it the format
specifier, a `%s`, was not removed).

A more significant change (that only requires minor changes in mkiocccentry
code) is that the tools `jstrencode(1)` and `jstrdecode(1)` were swapped as
according to other sources, converting a code point to another character
(accented character, emoji etc.) is encoding, not decoding. This means that
functions and comments were also changed. Only minor changes in this repo had to
be made, as noted, however and specifically function name changes.

With this merged the temp-test-ioccc repo will have to have the scripts updated
to use the right tool (not jstrdecode but jstrencode) and the correct version
(2.0.0).

## Release 1.6.2 2024-10-30

Sync the `jparse/` directory from [jparse
repo](https://github.com/xexyl/jparse/). This includes some important bug fixes,
especially in the `make install` rule that did not install a header file. Not
having this would cause a compilation error if one were to try and include
`jparse/jparse.h` outside this repo.

The `jparse_bug_report.sh` script has an improvement that will test compile a
jparse program to see if the system can link in `libjparse.a`, `libdbg.a` and
`libdyn_array.a`. This script is not needed for this repo though.

## Release 1.6.2 2024-10-23

Fix annoying bug where the `MKIOCCCENTRY_ANSWERS_VERSION` could not be in
`soup/version.h` and the `MKIOCCCENTRY_ANSWERS_EOF` could not be in
`mkiocccentry.h`.

Fix `write_echo()` in `test_ioccc/prep.sh` to not leave lines in the log file
that simply say `OK`.

## Release 1.6.1 2024-10-22

Sync the `jparse/` directory with state, as of 2024-10-22, of the
[jparse repo](https://github.com/xexyl/jparse/) .  Those updates
include improvements to the jparse test suite including new test cases,
test suite improvements, and related man pages.  The `jstrencode(1)`,
`jstrdecode(1)` and related Unicode / UTF-8 functions have been improved.

Changed `MKIOCCCENTRY_ANSWERS_VERSION` from "MKIOCCCENTRY_ANSWERS-IOCCCMOCK-1.0"
to "MKIOCCCENTRY_ANSWERS_IOCCC28-1.0".

Changed `MKIOCCCENTRY_REPO_VERSION` from "1.6 2024-10-10"
to 1.6.1 2024-10-22.

<hr /><em>This discussion was created from the release <a href='https://github.com/ioccc-src/mkiocccentry/releases/tag/v2.0'>Release version 2.0 2024-11-17</a>.</em>

**Comment by lcn2** [comment]
# QUESTION 

Is there a way for GitHub to create, automatically, a .tar.gz and a .zip file for a given release?

In the case of the [calc repo](https://github.com/lcn2/calc/releases/tag/v2.15.1.0) (see the Assets section at the bottom of that page) when we upload RPM files, GitHub generates both a .tar.gz and a .zip file.  

Is there a way that GitHub could generate both a .tar.gz and a .zip file for a given new release without of having to upload some files?

**Reply by xexyl** [reply]
> # QUESTION 
> 
> Is there a way for GitHub to create, automatically, a .tar.gz and a .zip file for a given release?
> 
> In the case of the [calc repo](https://github.com/lcn2/calc/releases/tag/v2.15.1.0) (see the Assets section at the bottom of that page) when we upload RPM files, GitHub generates both a .tar.gz and a .zip file.  
> 
> Is there a way that GitHub could generate both a .tar.gz and a .zip file for a given new release without of having to upload some files?

I thought that it does that automatically. Or are you saying you want one for each release?

Wearing distance glasses and as noted elsewhere today is not good (although I am feeling better mostly and will be fine) and additionally I am just passing by so if you need to clarify please do. 

I guess that there is not a way to do what I think you're asking but I don't know for sure .. but of course I also don't know if I am parsing what you're asking correctly.

**Reply by xexyl** [reply]
Btw what I am referring to: if you go to the repo's main page there's a link that allows one to download it through afaik it's only a zip file.

**Reply by xexyl** [reply]
Seems like that feature might only be available in web browsers but I don't know. I don't see it in the app at a quick blindish look.

**Reply by lcn2** [reply]
> Seems like that feature might only be available in web browsers but I don't know. I don't see it in the app at a quick blindish look. 

We don't see the link on an iPhone.  Perhaps this is a GitHub "feature" of iPhones that they don't supply a zip and/or tarball?

**Reply by xexyl** [reply]
> > Seems like that feature might only be available in web browsers but I don't know. I don't see it in the app at a quick blindish look.
> 
> We don't see the link on an iPhone. Perhaps this is a GitHub "feature" of iPhones that they don't supply a zip and/or tarball?

I didn't see it there either. I only see it on 'desktop'.

**Reply by xexyl** [reply]
This is what it looks like after clicking on the down arrow:

![Screenshot 2024-11-19 at 09 49 44](https://github.com/user-attachments/assets/e2ba7601-d548-42ac-8923-d69702538a22)

**Comment by lcn2** [comment]
OK, the concern raised with [GH-discussioncomment](https://github.com/ioccc-src/mkiocccentry/discussions/1030#discussioncomment-11287052) is **NOT** a problem as it only impacts mobile devices and not the desktop which **DOES** have the proper zip and tarball links.

**Reply by xexyl** [reply]
Correct.
